test/adapter_test.rb

```diff
@@ -0,0 +1,58 @@
+require 'test_helper'
+
+class TestSemianAdapter < Minitest::Test
+  def setup
+    Semian.unregister_all_resources
+    # Consumers registered in other test files must be cleared
+    Semian.instance_variable_set(:@consumers, nil)
+    GC.enable
+    GC.start
+  end
+
+  def teardown
+    Semian.unregister_all_resources
+  end
+
+  def test_adapter_registers_consumer
+    assert_equal(true, Semian.resources.empty?)
+    assert_equal(true, Semian.consumers.empty?)
+    client = Semian::AdapterTestClient.new(quota: 0.5)
+    resource = client.semian_resource
+    assert_equal(resource, Semian.resources[client.semian_identifier])
+    assert_equal(client.inspect, Semian.consumers[client.semian_identifier].first.inspect)
+  end
+
+  def test_unregister_all_resources
+    client = Semian::AdapterTestClient.new(quota: 0.5)
+    resource = client.semian_resource
+    assert_equal(resource, Semian.resources[client.semian_identifier])
+    assert_equal(client.inspect, Semian.consumers[client.semian_identifier].first.inspect)
+
+    GC.disable # need to disable GC to ensure client weak reference is alive for assertion below
+    Semian.unregister_all_resources
+
+    assert_equal(true, Semian.resources.empty?)
+    assert_equal(true, Semian.consumers.empty?)
+    assert_nil client.instance_variable_get(:@semian_resource)
+    GC.enable
+  end
+
+  def test_consumer_registration_does_not_prevent_gc
+    client = Semian::AdapterTestClient.new(quota: 0.5)
+    client.semian_resource
+    identifier = client.semian_identifier
+
+    client = nil
+    weak_ref = Semian.consumers[identifier].first
+    assert_equal(true, weak_ref.weakref_alive?)
+
+    GC.start
+
+    assert_nil client
```

Comment:
just so rubocop didn't yell at me for setting it to nil and not using it :| bad form i guess.